substratest/factory.py

```diff
@@ -221,11 +222,24 @@ class DataSampleSpec(_Spec):
     test_only: bool
     data_manager_keys: typing.List[str]
 
-    def move_data_to(self, destination):
-        destination = destination if destination.endswith('/') else destination + '/'
-        destination = tempfile.mkdtemp(dir=destination)
-        shutil.move(self.path, destination)
-        self.path = os.path.join(destination, os.path.basename(self.path))
+    def move_data_to(self, copy_destination, minikube=False):
+        copy_destination = copy_destination if copy_destination.endswith('/') else copy_destination + '/'
+
+        if not minikube:
+            destination = tempfile.mkdtemp(dir=copy_destination)
+            shutil.move(self.path, destination)
+        else:
+            destination = os.path.join(copy_destination, random_uuid()[0:8])
+
+            minikube_private_key = '~/.minikube/machines/minikube/id_rsa'
+            minikube_ssh = 'docker@$(minikube ip)'
+
+            os.system(f'scp -r -i {minikube_private_key} {self.path} {minikube_ssh}:/tmp/')
+            os.system(f'ssh -i {minikube_private_key} {minikube_ssh} '
+                      f'"sudo mkdir -p {destination} && sudo mv /tmp/{os.path.basename(self.path)} {destination}"')
```

Comment:
Could we do that in a single scp command?

If I understand well the ssh is required for the directory creation. 

Could we use instead an scp option (it looks like the `-r` option suits our need: `-r      Recursively copy entire directories.  Note that scp follows symbolic links encountered in the tree traversal.`) ?

Reply:
The `-r` option is already used (l.237)
With scp or ssh with private key, we use only an user account and we need `root` permissions to write data in `{destination}` path (hostpath volume in minikube is under root user)

Reply:
Good point, thanks for the explanation. 

What do you think of doing first the mkdir, then a chown operation, and to finish the scp to the final destination? 

It would avoid the `mv` operation.

Reply:
I tried to do that, but hostPath need to be root for k8s pods in minikube if we don't want to have permission issues inside :)

Reply:
Ok, thanks :)